### ISSUE TITLE
[Bug]: google-cloud-vision 3.x doesn't work in Beam

### ISSUE THREAD

**Issue by TheNeuralBit** [apache/beam#23261]
### What happened?

See https://github.com/apache/beam/pull/23239 which attempted to perform the upgrade. We get the error:
```
apache_beam/ml/gcp/visionml_test.py:36: in <module>
    from apache_beam.ml.gcp import visionml
apache_beam/ml/gcp/visionml.py:57: in <module>
    class AnnotateImage(PTransform):
apache_beam/ml/gcp/visionml.py:156: in AnnotateImage
    Union[str, bytes], Optional[vision.types.ImageContext])
E   AttributeError: module 'google.cloud.vision' has no attribute 'types'
```

### Issue Priority

Priority: 2

### Issue Component

Component: io-py-gcp

**Comment by johnjcasey** [comment]
@tvalentyn another example of version pinning issues

**Comment by TheNeuralBit** [comment]
In this case I think our system is working as intended, the <4 guard prevents users from getting the newer version that will break them.

**Comment by tvalentyn** [comment]
@AnandInguva  can you PTAL to see if we can update the Beam code to support newer version of this dependency?

**Comment by tvalentyn** [comment]
Sample fix: #23365 + #23480

**Comment by AnandInguva** [comment]
.take-issue
